cmd/revad/runtime/drivenserver.go

```diff
@@ -0,0 +1,181 @@
+package runtime
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/opencloud-eu/reva/v2/pkg/registry"
+	"github.com/rs/zerolog"
+)
+
+const (
+	HTTP = iota
+	GRPC
+)
+
+// RevaDrivenServer is an interface that defines the methods for starting and stopping reva HTTP/GRPC services.
+type RevaDrivenServer interface {
+	Start() error
+	Stop() error
+}
+
+// revaServer is an interface that defines the methods for starting and stopping a reva server.
+type revaServer interface {
+	Start(ln net.Listener) error
+	Stop() error
+	GracefulStop() error
+	Network() string
+	Address() string
+}
+
+// sever represents a generic reva server that implements the RevaDrivenServer interface.
+type server struct {
+	srv                     revaServer
+	log                     *zerolog.Logger
+	gracefulShutdownTimeout time.Duration
+	protocol                string
+}
+
+// NewDrivenHTTPServerWithOptions runs a revad server w/o watcher with the given config file and options.
+// Use it in cases where you want to run a revad server without the need for a watcher and the os signal handling as a part of another runtime.
+// Returns nil if no http server is configured in the config file.
+// The GracefulShutdownTimeout set to default 20 seconds and can be overridden in the core config.
+// Logging a fatal error and exit with code 1 if the http server cannot be created.
+func NewDrivenHTTPServerWithOptions(mainConf map[string]interface{}, opts ...Option) RevaDrivenServer {
+	if !isEnabledHTTP(mainConf) {
+		return nil
+	}
+	options := newOptions(opts...)
+	if srv := newServer(HTTP, mainConf, options); srv != nil {
+		return srv
+	}
+	options.Logger.Fatal().Msg("nothing to do, no http enabled_services declared in config")
```

Comment:
The log message seems misleading. If there were no http enabled service in the config we'd already returned on line 50. 

This line in only reached if `newServer` for whatever reason fails to initialize the server.